File: src/main/java/no/nav/bidrag/commons/web/HttpStatusResponse.java
package no.nav.bidrag.commons.web;

import java.util.Optional;
import org.springframework.http.HttpStatus;

public class HttpStatusResponse<T> {

  private final HttpStatus httpStatus;
  private final T body;

  public HttpStatusResponse(HttpStatus httpStatus) {
    this(httpStatus, null);
  }

  public HttpStatusResponse(HttpStatus httpStatus, T body) {
    this.httpStatus = httpStatus;
    this.body = body;
  }

  public Optional<T> fetchOptionalResult() {
    return Optional.ofNullable(body);
  }

  public boolean isNotSuccessful() {
    return !httpStatus.is2xxSuccessful();
  }

  @Override
  public String toString() {
    return "HttpStatusResponse{" +
        "httpStatus=" + httpStatus +
        ", body=" + body +
        '}';
  }

  public HttpStatus getHttpStatus() {
    return httpStatus;
  }

  public boolean isBodyEmpty() {
    return body == null;
  }

  public boolean isBodyPresent() {
    return body != null;
  }

  public T getBody() {
    return body;
  }
}

File: src/main/kotlin/no/nav/bidrag/commons/web/HttpStatusResponse.kt
package no.nav.bidrag.commons.web

import org.springframework.http.HttpStatus
import java.util.Optional

@Deprecated("")
class HttpStatusResponse<T : Any>
    @JvmOverloads
    constructor(val httpStatus: HttpStatus, val body: T? = null) {
        fun fetchOptionalResult(): Optional<T> {
            return Optional.ofNullable(body)
        }

        val isNotSuccessful: Boolean
            get() = !httpStatus.is2xxSuccessful

        override fun toString(): String {
            return "HttpStatusResponse{httpStatus=$httpStatus, body=$body}"
        }

        val isBodyEmpty: Boolean
            get() = body == null
        val isBodyPresent: Boolean
            get() = body != null
    }
